packages/create-amplify/src/package-manager-controller/package_manager_controller_factory.ts

```diff
@@ -0,0 +1,38 @@
+import { execa as _execa } from 'execa';
+import { type PackageManagerName } from './package_manager.js';
+import { NpmPackageManagerController } from './npm_package_manager_controller.js';
+import { PnpmPackageManagerController } from './pnpm_package_manager_controller.js';
+import { YarnClassicPackageManagerController } from './yarn_classic_package_manager_controller.js';
+import { YarnModernPackageManagerController } from './yarn_modern_package_manager_controller.js';
+
+export type DependencyType = 'dev' | 'prod';
+
+/**
+ *
+ */
+export abstract class PackageManagerController {
+  abstract installDependencies: (
+    packageNames: string[],
+    type: DependencyType
+  ) => Promise<void>;
+}
+
+/**
+ * packageManagerControllerFactory creates PackageManagerController for the Package Manager
+ */
+export const packageManagerControllerFactory = (
+  packageManagerName: PackageManagerName
+) => {
+  switch (packageManagerName) {
+    case 'npm':
+      return NpmPackageManagerController;
```

Comment:
1. factory should return object instance, not constructor.
2. If we pursue "static factory method" pattern then this function should be named `getPackageManagerControllerInstance()` or so. Functions/methods should have names that read like verb. Nouns are reserved for type/class/property names.

Reply:
Done 11796e3a5